src/iOS.Core/Controllers/LoginPasswordViewController.cs

```diff
@@ -0,0 +1,218 @@
+﻿using System;
+using UIKit;
+using Foundation;
+using Bit.iOS.Core.Views;
+using Bit.App.Resources;
+using Bit.iOS.Core.Utilities;
+using Bit.App.Abstractions;
+using Bit.Core.Abstractions;
+using Bit.Core.Utilities;
+using System.Threading.Tasks;
+using Bit.Core.Models.Domain;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+namespace Bit.iOS.Core.Controllers
+{
+    public abstract class LoginPasswordViewController : ExtendedUITableViewController
+    {
+        private IDeviceActionService _deviceActionService;
+        private IStorageService _storageService;
+        private IAuthService _authService;
+        private ISyncService _syncService;
+
+        public LoginPasswordViewController(IntPtr handle)
+            : base(handle)
+        { }
+
+        public abstract UINavigationItem BaseNavItem { get; }
+        public abstract UIBarButtonItem BaseCancelButton { get; }
+        public abstract UIBarButtonItem BaseSubmitButton { get; }
+        public abstract Action Success { get; }
+        public abstract Action Cancel { get; }
+
+        public FormEntryTableViewCell EmailCell { get; set; } = new FormEntryTableViewCell(
+            AppResources.EmailAddress);
+        public FormEntryTableViewCell MasterPasswordCell { get; set; } = new FormEntryTableViewCell(
+            AppResources.MasterPassword);
+
+        public override void ViewDidLoad()
+        {
+            _deviceActionService = ServiceContainer.Resolve<IDeviceActionService>("deviceActionService");
+            _storageService = ServiceContainer.Resolve<IStorageService>("storageService");
+            _authService = ServiceContainer.Resolve<IAuthService>("authService");
+            _syncService = ServiceContainer.Resolve<ISyncService>("syncService");
+
+            BaseNavItem.Title = AppResources.Bitwarden;
+            BaseCancelButton.Title = AppResources.Cancel;
+            BaseSubmitButton.Title = AppResources.LogIn;
+
+            var descriptor = UIFontDescriptor.PreferredBody;
+            
+            MasterPasswordCell.TextField.SecureTextEntry = true;
+            MasterPasswordCell.TextField.ReturnKeyType = UIReturnKeyType.Go;
+            MasterPasswordCell.TextField.ShouldReturn += (UITextField tf) =>
+            {
+                // TODO This isn't working - bricks application
+                LogInAsync().GetAwaiter().GetResult();
```

Comment:
This pattern will commonly cause a deadlock issue with thread blocking in .net. To properly call an async method syncronously, the method body being called (and any downstream methods) has to play along as well using `.ConfigureAwait(false)` on the return `await Task` result. You'll run into Exception unwrapping problems as well in this context (so basically it could be failing/throwing exceptions but you'll never know it and it'll never be logged because it's in an invalid thread-state).

If absolutely necessary you can also use the [nested message loop hack](https://docs.microsoft.com/en-us/archive/msdn-magazine/2015/july/async-programming-brownfield-async-development#the-nested-message-loop-hack).

*ORrrrr*, it looks like there's a `.RunSynchronously()` method for `Task` and `Task<T>` in dotnet core 3.1 (from the docs): https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.runsynchronously?view=netcore-3.1

Reply:
Thanks! Since I don't really need to wait for the result of this method, I've updated it to remove the Awaiter/Result pattern above. It seems to be working as intended now.